apps/api-gateway/src/user/user.controller.ts

```diff
@@ -222,37 +122,58 @@ export class UserController {
 
   }
 
-  @Get('public-profile')
+  @Get('/users/activity')
   @ApiOperation({
-    summary: 'Fetch user details',
-    description: 'Fetch user details'
-  })
-  @ApiQuery({
-    name: 'id',
-    type: Number,
-    required: false
+    summary: 'organization invitations',
+    description: 'Fetch organization invitations'
   })
-  async getPublicProfile(@User() reqUser: user, @Query('id') id: number, @Res() res: Response): Promise<object> {
-    const userData = await this.userService.getPublicProfile(id);
+  @UseGuards(AuthGuard('jwt'))
+  @ApiBearerAuth()
+  @ApiQuery({ name: 'limit', required: true })
+  @Roles(OrgRoles.OWNER, OrgRoles.SUPER_ADMIN, OrgRoles.ADMIN, OrgRoles.ISSUER, OrgRoles.VERIFIER, OrgRoles.MEMBER)
+  async getUserActivities(@Query('limit') limit: number, @Res() res: Response, @User() reqUser: user): Promise<Response> {
+
+    const userDetails = await this.userService.getUserActivities(reqUser.id, limit);
 
     const finalResponse: IResponseType = {
       statusCode: HttpStatus.OK,
-      message: ResponseMessages.user.success.fetchProfile,
-      data: userData.response
+      message: 'User activities fetched successfully',
+      data: userDetails.response
     };
 
     return res.status(HttpStatus.OK).json(finalResponse);
-
   }
 
-  @Get('invitations')
+
+  @Get('/users/org-invitations')
```

Comment:
If `/users` is repeating for all the APIs. you can set it to the top inside `@Controllers('users')`